File: lib/py/uapi/internal/schema/GetTypeUnexpectedParseFailure.py
from typing import list, object, dict

from uapi.internal.schema.SchemaParseFailure import SchemaParseFailure


def get_type_unexpected_parse_failure(path: list[object], value: object, expected_type: str) -> list[SchemaParseFailure]:
    from uapi.internal.types.GetType import get_type

    actual_type = get_type(value)
    data = {
        "actual": {actual_type: {}},
        "expected": {expected_type: {}}
    }
    return [
        SchemaParseFailure(path, "TypeUnexpected", data, None)
    ]

File: lib/ts/uapi/internal/schema/GetTypeUnexpectedParseFailure.ts
import { SchemaParseFailure } from 'uapi/internal/schema/SchemaParseFailure';
import { getType } from 'uapi/internal/types/GetType';

export function getTypeUnexpectedParseFailure(path: any[], value: any, expectedType: string): SchemaParseFailure[] {
    const actualType = getType(value);
    const data: { actual: { [key: string]: any }; expected: { [key: string]: any } } = {
        actual: { [actualType]: {} },
        expected: { [expectedType]: {} },
    };
    return [new SchemaParseFailure(path, 'TypeUnexpected', data, null)];
}
